app/assets/javascripts/discourse/app/controllers/flag.js

```diff
@@ -133,9 +185,28 @@ export default Controller.extend(ModalFunctionality, {
       }
     },
 
-    takeAction() {
-      this.send("createFlag", { takeAction: true });
-      this.set("model.hidden", true);
+    takeAction(action) {
+      let performAction = (o = {}) => {
+        o.takeAction = true;
+        this.send("createFlag", o);
+        return Promise.resolve();
```

Comment:
what is the goal of this promise here ?

Reply:
The promise is necessary for silence/suspend. It's used in a callback in the `before()` option for penalty-controller.js, called here - https://github.com/discourse/discourse/blob/master/app/assets/javascripts/admin/addon/mixins/penalty-controller.js#L57